include/configs/pistachio_bub.h

```diff
@@ -307,18 +305,25 @@
 
 #endif
 
+/*
+ * openwrt kernel has patch which automatically create rootfs, it
+ * fails if nandroot is passed to it. Assume openwrt uses kernel
+ * volume, so dont append nandroot to bootargs when booting from kernel
+ * volume. For other cases append nandroot to bootargs
+ */
 #define NAND_BOOTCOMMAND	\
-	"setenv nandroot ubi.mtd=firmware$boot_partition $nandroot;"\
-	"setenv bootargs $console $earlycon $nandroot $bootextra $mtdparts panic=2;"\
 	"echo Attempting to boot from firmware$boot_partition;"\
+	"setenv ubimtd ubi.mtd=firmware$boot_partition;"\
 	"ubi part firmware$boot_partition || reset;"\
 	"if ubi check kernel; then "\
 		"echo Loading kernel from volume...;"\
+		"setenv bootargs $console $earlycon $ubimtd $bootextra $mtdparts panic=2;"\
 		"ubi read $loadaddr kernel || reset;"\
 	"else "\
 		"echo Loading kernel from rootfs...;"\
+		"setenv bootargs $console $earlycon $ubimtd $nandroot $bootextra $mtdparts panic=2; && "\
```

Comment:
This will break Ci40 legacy boot for Openwrt which is out there in the field and therefore a mandatory requirement. Can't let this in as it would risk leaving public users with bricked boards. I'm not sure is going to be possible to support both, this way.